Pod/Classes/CDTQQueryValidator.m

```diff
@@ -112,6 +136,102 @@ + (NSArray *)addImplicitEq:(NSArray *)andClause
     return [NSArray arrayWithArray:accumulator];
 }
 
++ (NSArray *)compressMultipleNotOperators:(NSArray *)clause
+{
+    NSMutableArray *accumulator = [NSMutableArray array];
+    
+    for (NSDictionary *fieldClause in clause) {
+        NSObject *predicate = nil;
+        NSString *fieldName = nil;
+        // if this isn't a dictionary, we don't know what to do so pass it back
+        if ([fieldClause isKindOfClass:[NSDictionary class]] && [fieldClause count] != 0) {
+            fieldName = fieldClause.allKeys[0];
+            predicate = fieldClause[fieldName];
+        } else {
+            [accumulator addObject:fieldClause];
+            continue;
+        }
+        
```

Comment:
I agree that the comment could use more detail.  This comment/code block was lifted from https://github.com/cloudant/CloudantQueryObjc/commit/93e685412415722e5ad741279c2b3989e2f8fe15#diff-ebadd34f17d6c735a0858621573aad3fR113 and reused.  It applies in both cases.  I will add more detail to both comments (Actually all three comments since it is used once again a few lines below).